File: backend/app/schemas/report.py
from datetime import datetime

from pydantic import BaseModel


class SalesPoint(BaseModel):
    label: str
    value: float


class TopSellingItem(BaseModel):
    menu_item_id: int
    name: str
    quantity_sold: int
    revenue: float


class SalesReport(BaseModel):
    start_date: datetime | None = None
    end_date: datetime | None = None
    total_orders: int
    paid_orders: int
    total_revenue: float
    average_order_value: float
    by_payment_method: dict[str, float]
    by_status: dict[str, int]
    sales_over_time: list[SalesPoint] = []
    top_items: list[TopSellingItem] = []

File: app/src/types/api/report.ts
export type ReportPeriod = "today" | "week" | "month";

export type SalesReportApi = {
	start_date: string | null;
	end_date: string | null;
	total_orders: number;
	paid_orders: number;
	total_revenue: number;
	average_order_value: number;
	by_payment_method: Record<string, number>;
	by_status: Record<string, number>;
	sales_over_time: { label: string; value: number }[];
	top_items: {
		menu_item_id: number;
		name: string;
		quantity_sold: number;
		revenue: number;
	}[];
};
